### PR TITLE
test(upgrade): remove `setTimeout` from lifecycle hook tests 

### LINKED ISSUE #13019: Upgrade tests are flaky
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
```

**Current behavior**

Some of the upgrade tests are flaky on BrowserStack due to the use of `setTimeout()`.

**Expected behavior**

The upgrade tests should not be flaky.

**Minimal reproduction of the problem with instructions**

See: https://travis-ci.org/angular/angular/jobs/177320560

The test fails in the above but it will periodically succeed if the test is re-run.

### PR DESCRIPTION
**Pull request by petebacondarwin**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [N/A] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

Some tests were flaky due to use of `setTimeout`

**What is the new behavior?**

The tests do not now rely upon `setTimeout` so "should" be less flaky.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Fixes #13019

### CODE REVIEW

**Comment by petebacondarwin** [comment]
@bradyisom can you take a look?

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 1]
```diff
@@ -892,154 +873,155 @@ export function main() {
...
+           const EXPECTED_VALUE = '$onChanges called';
            const adapter: UpgradeAdapter = new UpgradeAdapter(forwardRef(() => Ng2Module));
-           const ng1Module = angular.module('ng1', []);
-           const valueToFind = '$onChanges init';
-           const valueToChange = '$onChanges changed';
+           const $onChangesSpy = jasmine.createSpy('$onChanges');
+           let ng2Instance: any;
+
+           @Component({selector: 'ng2', template: '<ng1 [val]="val"></ng1>'})
+           class NG2Component {
```
Comment (gkalpak) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: Why not `Ng2Component` (lowercase `g`)? :grin:

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 2]
```diff
@@ -892,154 +873,155 @@ export function main() {
...
            });
          }));
 
-      it('should call $onDestroy of components', async(() => {
+      it('should call $onDestroy of components', fakeAsync(() => {
            const adapter: UpgradeAdapter = new UpgradeAdapter(forwardRef(() => Ng2Module));
-           const ng1Module = angular.module('ng1', []);
 
-           let spy = jasmine.createSpy('$onDestroy');
+           let $onDestroySpy = jasmine.createSpy('$onDestroy');
```
Comment (gkalpak) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: `let` --> `const`

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 3]
```diff
@@ -892,154 +873,155 @@ export function main() {
...
-                 expect(spy).toHaveBeenCalled();
-                 ref.dispose();
-               }, 100);
-             }, 100);
+             const $rootScope = ref.ng1RootScope as any;
+
+             $rootScope.destroy = false;
+             tick();
+             $rootScope.$digest();
+
```
Comment (gkalpak) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: Let's add a assertion that `$onDestroySpy` hasn't been called here.

**Review by gkalpak — APPROVED** [review]
LGTM (with a couple of minor suggestions)

**Comment by bradyisom** [comment]
LGTM

**Pull request merged**